### PR TITLE
Introduce client_id to the oidc/logout endpoint

### PR DESCRIPTION
**Pull request by Yasasr1**
### Proposed changes in this pull request

This PR introduces an additional optional parameter `client_id` to the oidc/logout endpoint. This parameter should contain the client_id of the RP application that initiated the logout request. This parameter can be used instead of `id_token_hint parameter`

#### Related issue
- https://github.com/wso2/product-is/issues/16671

#### Integration test PR
- https://github.com/wso2/product-is/pull/16746

### CODE REVIEW

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java [hunk 1]
```diff
@@ -325,32 +326,33 @@ private String processLogoutRequest(HttpServletRequest request, HttpServletRespo
...
-                appTenantDomain = request.getParameter(OIDCSessionConstants.OIDC_TENANT_DOMAIN_PARAM);
-                JWT decryptedIDToken = OIDCSessionManagementUtil.decryptWithRSA(appTenantDomain, idTokenHint);
-                clientId = OIDCSessionManagementUtil.extractClientIDFromDecryptedIDToken(decryptedIDToken);
-            } else {
-                if (!validateIdToken(idTokenHint)) {
-                    String msg = "ID token signature validation failed.";
-                    if (log.isDebugEnabled()) {
-                        log.debug(msg);
+            if (StringUtils.isEmpty(clientId)) {
+                if (OIDCSessionManagementUtil.isIDTokenEncrypted(idTokenHint)) {
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java:338]: shall we extract this part to a private method. `getClientIdFromIdToken`

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java [hunk 2]
```diff
@@ -325,32 +326,33 @@ private String processLogoutRequest(HttpServletRequest request, HttpServletRespo
...
-            if (OIDCSessionManagementUtil.isIDTokenEncrypted(idTokenHint)) {
-                appTenantDomain = request.getParameter(OIDCSessionConstants.OIDC_TENANT_DOMAIN_PARAM);
-                JWT decryptedIDToken = OIDCSessionManagementUtil.decryptWithRSA(appTenantDomain, idTokenHint);
-                clientId = OIDCSessionManagementUtil.extractClientIDFromDecryptedIDToken(decryptedIDToken);
-            } else {
-                if (!validateIdToken(idTokenHint)) {
-                    String msg = "ID token signature validation failed.";
-                    if (log.isDebugEnabled()) {
-                        log.debug(msg);
+            if (StringUtils.isEmpty(clientId)) {
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java:337]:
```suggestion
            if (StringUtils.isBlank(clientId)) {
```

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java [hunk 3]
```diff
@@ -380,7 +382,9 @@ private String processLogoutRequest(HttpServletRequest request, HttpServletRespo
         paramMap.put(OIDCSessionConstants.OIDC_CACHE_CLIENT_ID_PARAM, clientId);
         paramMap.put(OIDCSessionConstants.OIDC_CACHE_TENANT_DOMAIN_PARAM, appTenantDomain);
         OIDCSessionDataCacheEntry cacheEntry = new OIDCSessionDataCacheEntry();
-        cacheEntry.setIdToken(idTokenHint);
+        if (StringUtils.isNotEmpty(idTokenHint)) {
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java:385]:
```suggestion
        if (StringUtils. isNotBlank(idTokenHint)) {
```

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java (hunk 2 above)
Comment (janakamarasena) [reply components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java:337]: Check other relevant places and switch to isBlank as well

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java (hunk 1 above)
Comment (Yasasr1) [reply components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java]: addressed with [054f072](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2169/commits/054f07256bc9967336c3e5c7d9b3315ab85184b2)

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java (hunk 2 above)
Comment (Yasasr1) [reply components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java]: addressed with [054f072](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2169/commits/054f07256bc9967336c3e5c7d9b3315ab85184b2)

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java (hunk 3 above)
Comment (Yasasr1) [reply components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java]: addressed with [054f072](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2169/commits/054f07256bc9967336c3e5c7d9b3315ab85184b2)

**Review by janakamarasena — DISMISSED** [review]

**Review by janakamarasena — APPROVED** [review]

**Pull request merged**